lib/msf/core/modules/metadata/search.rb

```diff
@@ -9,11 +9,61 @@ module Msf::Modules::Metadata::Search
       %w[aka author authors arch cve bid edb check date disclosure_date description fullname fullname mod_time
       name os platform path port rport rank ref ref_name reference references target targets text type]
 
+  #
+  # Module Type Shorthands
+  #
+  MODULE_TYPE_SHORTHANDS = {
+    "aux" => Msf::MODULE_AUX
+  }
+
+  #
+  # Parses command line search string into a hash. A param prefixed with '-' indicates "not", and will omit results
+  # matching that keyword. This hash can be used with the find command.
+  #
+  # Resulting Hash Example:
+  # {"platform"=>[["android"], []]} will match modules targeting the android platform
+  # {"platform"=>[[], ["android"]]} will exclude modules targeting the android platform
+  #
+  def self.parse_search_string(search_string)
```

Comment:
This is the existing parse string functionality lifted from `command_dispatcher/modules.rb`